crates/core/src/host/instance_env.rs

```diff
@@ -144,23 +145,24 @@ impl InstanceEnv {
     ///
     /// Returns an error if no columns were deleted or if the column wasn't found.
     #[tracing::instrument(skip(self, value))]
-    pub fn delete_by_col_eq(&self, table_id: u32, col_id: u32, value: &[u8]) -> Result<u32, NodesError> {
+    pub fn delete_by_col_eq(&self, table_id: u32, col_id: u32, value: &[u8]) -> Result<NonZeroU32, NodesError> {
```

Comment:
I don't understand what removing clones has to do with the type of this function.

Reply:
I'm fine with this change for now. I'm surprised at these semantics, though, but I agree that the code and the docs all agreeing right now is an improvement.